packages/api/cms-api/src/dam/files/file-validation-interceptor.service.ts

```diff
@@ -0,0 +1,36 @@
+import { CallHandler, ExecutionContext, HttpException, Injectable, NestInterceptor } from "@nestjs/common";
+import fs from "fs";
+import { Observable, throwError } from "rxjs";
+import * as util from "util";
+
+import { svgContainsJavaScript } from "./files.utils";
+
+const readFile = util.promisify(fs.readFile);
+const unlinkFile = util.promisify(fs.unlink);
+
+@Injectable()
+export class FileValidationInterceptor implements NestInterceptor {
```

Comment:
Shouldn't this validation happen in the DamUploadFileInterceptor? This would prevent uploading the file at all

Reply:
I moved the validation to the DamUploadFileInterceptor. 

However, it's not possible to do it before the upload happens because I can't access the file content in the fileFilter: 
<img width="314" alt="Bildschirm­foto 2023-08-29 um 09 22 21" src="https://github.com/vivid-planet/comet/assets/13380047/0b989f07-5464-43d2-88e2-e900d1b622a2">

I couldn't find a way to access the content as a buffer before upload.

Reply:
When thinking about it, this probably wasn't a good suggestion, as it now doesn't validate uploaded files in the public uploads module. Could you please revert this change? Sorry 😬

Reply:
Okay 👍